Source/Core/DolphinQt2/Config/Graphics/GeneralWidget.cpp

```diff
@@ -299,12 +299,14 @@ void GeneralWidget::AddDescriptions()
                  "cost of high GPU requirements. Only recommended for high-end systems.");
   static const char* TR_SHADER_COMPILE_ASYNC_UBER_DESCRIPTION =
       QT_TR_NOOP("Ubershaders will be used to prevent stuttering during shader compilation, but "
-                 "specialized shaders will be used when they will not cause stuttering.");
+                 "specialized shaders will be used when they will not cause stuttering. Balances "
+                 "GPU strain and stutter reduction.");
```

Comment:
> Balances GPU strain and stutter reduction.

That's actually not really true. It's not a "balance" at all. In cases where Async Ubershaders works, there is no downside, it's just ubershaders without the performance hit. It's awesome! The problem with Async ubershaders is that it is pretty demanding on driver features, so it can have stuttering in various ways due to how the drivers deal with them. I'd go for something like this here:

> In the best case it is Ubershaders with nearly no performance impact, but that depends on video drivers.

Reply:
Ok, that went unnoticed in the update Android Ubershaders description PR.

Reply:
Since synchronous (low-end systems) and synchronous ubershaders (high-end systems) mention hardware requirements, I think asynchronous ubershaders should mention it too. Also, isn't there still more stutter with Async Ubershaders vs Sync Ubershaders? It's definitely minimized but it's still noticeable if you look for it. How about:

If supported by video drivers, this is near Synchronous Ubershaders stutter reduction with minimal performance impact and less hardware requirements.

Reply:
Found it in #5702.
> Hybrid: Compiles specialized shaders asynchronously, while this occurs, ubershaders will be used. Best balance of performance and stuttering. This is not the same as the Ishiiruka async shaders! Game objects will continue to render as normal while shaders are being compiled. Not guaranteed to remove stuttering completely, as drivers often defer some work to the first time a program is used, and/or GL_LINK_STATUS is checked (cough NVIDIA), which we do on the main thread.

If the statement is correct for Async Ubershaders, I think I still like:
> If supported by video drivers, this is near Synchronous Ubershaders stutter reduction with minimal performance impact and less hardware requirements.

Reply:
Unfortunately it isn't really that simple. When it works, Async Ubershaders has zero stuttering, with no performance impact whatsoever. When it works It is THE BEST way to play Dolphin! But when the driver is unhappy with it, it stutters, and it's little better than Synchronous without ubershaders.

And we can't even isolate it to "better GPUs are better". A super powerful GPU can be fine with async ubershaders on one backend, and stutter on another backend. It's all about the drivers and quirks that they do in the very unusual load we put on them. It's finicky.

Reply:
Ok, how about:

> Provides Synchronous Ubershaders stutter reduction with minimal performance impact and less hardware requirements when fully supported by video drivers.

The "less hardware requirements" part is important since hardware requirements are compared in Sync vs Sync Ubershaders.

Reply:
It seems I've made some bad assumptions about Ubershaders!
I can run Async Ubershaders at 100% (60 FPS) with significant stutter reduction in Brawl. Sync Ubershaders run at 20-50%. I'm guessing that this is because the ubershaders aren't used often with Async Ubershaders. My laptop can't support Sync Ubershaders but Async ubershaders are a major improvement. I think we want to encourage both high-end and low-end users to try Async ubershaders. How about:

> In the best case it provides Sync Ubershaders stutter reduction with minimal performance impact on high-end systems, but that depends on video driver behavior. Also gives decent performance and stutter reduction on lower-end systems with games that use Ubershaders less often.

Reply:
Thought about this more. Generally people see the "high-end system" option as the best way to run Dolphin (Sync Ubershaders). I think we should say that Async Ubershaders are the best option instead of having users guess. How about:
> In the best case it provides Sync Ubershaders stutter reduction with minimal performance impact on high-end systems. Also gives decent stutter reduction and performance on lower-end systems with games that use Ubershaders less often. If supported by video driver behavior, use this option for the best Dolphin experience.

Reply:
Ok, thanks for clearing that up. There still needs to be a more explicit way to compare Sync vs. Sync Ubershaders vs. Async Ubershaders. Sync and Sync Ubershaders mention GPU requirements and the new Async Ubershaders description mentions performance. That should hopefully fix that for Async Ubershaders:

> In the best case it eliminates shader compilation stuttering yet has low GPU demands and performance impact, but results depends on video driver behavior.

Also, change Sync Ubershaders' "GPU requirements" to "GPU demands" for consistency and comparison aid.

These changes make it clear that GPU demands and stutter reduction are the factors that users need to consider when picking the best shader compilation option for them.

Reply:
Well, I kind of want Async to still say the word "performance", since that is its major benefit and I really want users to see that magic word here to entice them to use it. But I don't really want it mention GPU demands AND performance. Not only is that kind of inaccurate, as outside of performance needs the demands are the same between sync and async ubershaders, but it's also long and rather clunky.

So I'd honestly prefer keeping GPU requirements (or changing it to GPU demands, that's a nice word. it's smaller :3 ) for Sync Ubershaders, and having only "minimal performance impact" on Async Ubershaders. It's both concise and accurate that way.

So like:

>In the best case it eliminates shader compilation stuttering yet has minimal performance impact, but results depend on video driver behavior.

To maintain the same language but keep the word "performance" in Async, if we try to shift to an all performance perspective. It would be super clean in Async Ubershaders, but in Sync Ubershaders we'd need to do something like "demands high performance GPUs" or something and just, that's clumsy, not concise, and not very clear.

Using different terminology is really advantageous here, imo.

Reply:
I still dislike that a high-end system user will not be easily able to compare Sync Ubershaders vs Async Ubershaders from the descriptions. We want Async Ubershaders to sound more appealing than Sync Ubershaders so users try them first. Perhaps mention GPU demands and performance in Sync Ubershaders and go with your Async Ubershaders description? This adds minimal words but will let users know what to look for when comparing Sync Ubershaders vs Async Ubershaders.

Also, for Async Ubershaders, I think "while having" has a more positive connotation than "yet has" so I went with that.

For Sync Ubershaders:
> Ubershaders will always be used. Provides a near stutter-free experience at the cost of possible performance loss from high GPU requirements. Only recommended for high-end systems.

For Async Ubershaders:
> In the best case it eliminates shader compilation stuttering while having minimal performance impact, but results depend on video driver behavior.

Reply:
"possible performance loss" um, sync ubershaders is a HUGE GPU load. HUUUGE! Only high end systems can run it fullspeed, and even then, it requires reducing internal resolution and AA compared to what they could get with other options. It is a massive performance hit!

Also it's kind of clunky, in addition to being kind of wrong.... I still think the small wording difference is fine. So...

Since you really want the word performance to be in both for better comparisons, how about this? 

> Ubershaders will always be used. Provides a near stutter-free experience at the cost of high GPU performance requirements. Only recommended for high end systems.

And

> In the best case it eliminates shader compilation stuttering while having minimal performance impact, but results depend on video driver behavior.

I think having performance in both is unnecessary, but this setup doesn't hurt anything. If you really want word parity though, I'm cool with this!